From d77213e5f42b1050ee0c29fc53cd3c68b4d9cac2 Mon Sep 17 00:00:00 2001
From: Harish Rachakonda <quic_rachakon@quicinc.com>
Date: Mon, 6 May 2024 18:27:46 +0530
Subject: [PATCH] wifi: ath12k: Modify debug prints to display number of active vdevs

Modify debug prints to display number of active vdevs before
creating and deleting the vdev.

Patch-depdendency:none

Signed-off-by: Harish Rachakonda <quic_rachakon@quicinc.com>
---
 drivers/net/wireless/ath/ath12k/wmi.c                       | 6 ++++++
 1 file changed, 6 insertions(+), 0 deletions(-)

--- a/drivers/net/wireless/ath/ath12k/wmi.c
+++ b/drivers/net/wireless/ath/ath12k/wmi.c
@@ -1540,9 +1540,9 @@ int ath12k_wmi_vdev_create(struct ath12k
 	}
 
 	ath12k_dbg(ar->ab, ATH12K_DBG_WMI,
-		   "WMI vdev create: id %d type %d subtype %d macaddr %pM pdevid %d\n",
-		   args->if_id, args->type, args->subtype,
-		   macaddr, args->pdev_id);
+		   "WMI vdev create: id %d type %d subtype %d macaddr %pM pdevid %d vdev bitmap (allocate:%lld free:%lld)\n",
+		   args->if_id, args->type, args->subtype, macaddr, args->pdev_id,
+		   ar->allocated_vdev_map, ar->ab->free_vdev_map);
 
 	ret = ath12k_wmi_cmd_send(wmi, skb, WMI_VDEV_CREATE_CMDID);
 	if (ret) {
@@ -1570,8 +1570,10 @@ int ath12k_wmi_vdev_delete(struct ath12k
 						 sizeof(*cmd));
 	cmd->vdev_id = cpu_to_le32(vdev_id);
 
-	ath12k_dbg(ar->ab, ATH12K_DBG_WMI, "WMI vdev delete id %d num_peers : %d\n",
-		   vdev_id, ar->num_peers);
+	ath12k_dbg(ar->ab, ATH12K_DBG_WMI,
+		   "WMI vdev delete id %d num_peers : %d vdev bitmap (allocate:%lld free:%lld)\n",
+		   vdev_id, ar->num_peers, ar->allocated_vdev_map,
+		   ar->ab->free_vdev_map);
 
 	ret = ath12k_wmi_cmd_send(wmi, skb, WMI_VDEV_DELETE_CMDID);
 	if (ret) {
